misk-aws/src/main/kotlin/misk/jobqueue/sqs/AwsSqsQueueAttributeImporter.kt

```diff
@@ -0,0 +1,64 @@
+package misk.jobqueue.sqs
+
+import com.amazonaws.services.sqs.model.GetQueueAttributesRequest
+import com.amazonaws.services.sqs.model.QueueAttributeName
+import misk.clustering.lease.LeaseManager
+import misk.jobqueue.QueueName
+import misk.logging.getLogger
+import misk.tasks.RepeatedTaskQueue
+import misk.tasks.Status
+import java.time.Duration
+import javax.inject.Inject
+
+internal class AwsSqsQueueAttributeImporter @Inject constructor(
+  private val config: AwsSqsJobQueueConfig,
+  private val leaseManager: LeaseManager,
+  private val metrics: SqsMetrics,
+  private val queues: QueueResolver,
+  @ForSqsHandling private val taskQueue: RepeatedTaskQueue
+){
+
+  /**
+   * Spawn a new thread that will consume the [queueName] SQS attributes and record them as metrics
+   *
+   * The metric recorded are:
+   * - jobs_sqs_aproximate_number_of_messages
+   * - jobs_sqs_aproximate_number_of_messages_not_visible
+   */
+  fun import(queueName: QueueName) {
+    val frequency = Duration.ofMillis(config.queue_attribute_importer_frequency_ms)
+
+    taskQueue.scheduleWithBackoff(frequency) {
+      val queue = queues[queueName]
+      val lease = leaseManager.requestLease("sqs-queue-attributes-${queue.queueName}")
+      if (lease.checkHeld()) {
+        log.info {
```

Comment:
you probably want to clear the metrics if you don't own the lease, since they are gauges